src/python/pants/backend/python/subsystems/black.py

```diff
@@ -14,5 +14,7 @@ class Black(PythonToolBase):
   @classmethod
   def register_options(cls, register):
     super().register_options(register)
-    register('--config', advanced=True, type=file_option, fingerprint=True,
```

Comment:
`advanced` is a convention that we use for options that we expect that people will put in `pants.ini` rather than on the CLI. It's... probably not the clearest though.

Reply:
Hm. That does not seem very intuitive to me from a user's perspective. My thinking has always been `./pants help` is for the main things I'd want and `./pants help-advanced` is for super-user things like messing with the cache.

Do you believe the typical Pants user™️ knows this is what we mean between `./pants help` vs. `./pants help-advanced`?

Reply:
> My thinking has always been ./pants help is for the main things I'd want and ./pants help-advanced is for super-user things like messing with the cache.

I think that some of why this is blurry is that in a large monorepo, some of the users are maintainers of the build, and the rest are consumers. Each repository has different numbers of folks in each group. And the advanced flag is thus intended to ride that line: maintainers (ie, the people mostly responsible for editing `pants.ini`) vs users (mostly doing things on the CLI).

Reply:
Okay, I'll keep it at `advanced` then. Although I'll continue removing `fingerprint` because that's irrelevant in V2.